### PR TITLE
System tests for seq with package visibility changes.

### PR DESCRIPTION
**Pull request by rabbah**
System tests where entitlements change on referenced actions after creation.

### CODE REVIEW

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala [hunk 1]
```diff
@@ -86,7 +86,13 @@ class WskEntitlementTests
                 (wp, assetHelper) =>
                     assetHelper.withCleaner(wsk.action, fullyQualifiedActionName, confirmDelete = false) {
                         (action, name) =>
-                            val rr = action.create(name, None, shared = Some(true), update = true, expectedExitCode = FORBIDDEN)(wp)
+                            val rr = action.create(name, None, update = true, expectedExitCode = FORBIDDEN)(wp)
```
Comment (ioana-blue) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: could you please use var names that are more intuitive/less ambiguous? I'm guessing rr means "run result" and not "rodric rabbah". how about using "run" or "result"?

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala [hunk 2]
```diff
@@ -259,6 +265,63 @@ class WskEntitlementTests
             })(defaultWskProps)
     }
 
+    it should "invoke an action sequence from package" in withAssetCleaner(guestWskProps) {
```
Comment (ioana-blue) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: Not immediately obvious whether we need this test. We have this one: https://github.com/openwhisk/openwhisk/blob/master/tests/src/system/basic/WskSequenceTests.scala#L228

which tests a sequence that points to an action in a package (like the test under review); in addition, the existing test performs checks around parameter passing. Do we need this test you're adding?

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala [hunk 3]
```diff
@@ -259,6 +265,63 @@ class WskEntitlementTests
...
+
+                    val run = wsk.action.invoke("sequence")(defaultWskProps)
+                    withActivation(wsk.activation, run)({
+                        _.response.success shouldBe true
+                    })(defaultWskProps)
+            }
+    }
+
+    it should "not allow invoke an action sequence with more than one component from package after entitlement change" in withAssetCleaner(guestWskProps) {
+        (gwp, assetHelper) =>
```
Comment (ioana-blue) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: gwp = global wealth predictor? :) see comment above about rr.

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala [hunk 4]
```diff
@@ -259,6 +265,63 @@ class WskEntitlementTests
...
+            assetHelper.withCleaner(wsk.pkg, samplePackage) {
+                (pkg, _) =>
+                    pkg.create(samplePackage, parameters = Map("a" -> "A".toJson), shared = Some(true))(gwp)
+                    pkg.create(samplePackage + "prv", parameters = Map("a" -> "A".toJson), shared = Some(true))(gwp)
```
Comment (ioana-blue) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: factor out in a val `samplePackage + "prv"` ?

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala (hunk 1 above)
Comment (rabbah) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: we use `rr` in a few places in the test, it means run result, yes.

**Comment by rabbah** [comment]
pg1/936 approved.

**Comment by ioana-blue** [comment]
Rebased and squashed. Committed as 7ad4590

**Pull request closed without merging**